script.js

```diff
@@ -0,0 +1,99 @@
+var clicks = 0;
+
+var ticTacToe = {
+  boxSelector: $(".box"),
+  xClicked: [],
+  oClicked: [],
+  eventListener: function() {
+    //accepts the mouseclick/mouseover and adds an attribute to the HTML box classes to be selected later
+    for(var i = 0; i < this.boxSelector.length; i++){
+      this.boxSelector.eq(i).attr("data-index",i);
+      this.boxSelector[i].addEventListener("mouseover", this.mouseOver);
+      this.boxSelector[i].addEventListener("mouseout", this.mouseOut);
+      this.boxSelector.eq(i).on("click", this.mouseClick);
+    }
+  },
+  mouseClick: function(event) {
+    event.preventDefault();
+    var boxNumber = parseInt(this.getAttribute("data-index"));
+    if (clicks === 0 || ticTacToe.flipChecker(boxNumber)){
+      clicks++;
+      ticTacToe.turnChecker(boxNumber);
+    }
+  },
+  mouseOver: function(event) {
+    event.preventDefault();
+    var boxNumber = parseInt(this.getAttribute("data-index"));
+    ticTacToe.boxSelector.eq(boxNumber).css("backgroundColor", "gray");
+    if(ticTacToe.flipChecker(boxNumber)) {
+      if (clicks % 2 === 1) {
+        ticTacToe.boxSelector.eq(boxNumber).html("O");
+      } else if (clicks % 2 === 0) {
+        ticTacToe.boxSelector.eq(boxNumber).html("X");
+      }
+    }
+  },
+  mouseOut: function(event) {
+    event.preventDefault();
+    var boxNumber = parseInt(this.getAttribute("data-index"));
+    ticTacToe.boxSelector.eq(boxNumber).css("backgroundColor", "#282c37");
+
+  },
+  flipChecker: function(boxNumber){
+    var tileColor = ticTacToe.boxSelector.eq(boxNumber).css("color");
+    if(tileColor != "rgb(43, 144, 217)" && tileColor != "rgb(217, 225, 232)") {
+      return true;
+    } else {
+      return false;
+    }
```

Comment:
using a class might be a more reliable way to check and see if a box has been clicked. e.g. what if you decide to change the color scheme in the future? you will have to update your JS as well..